# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/gpu_info_util/SystemInfo.h
# Commit: 13282d7a47a5
# Full Hash: 13282d7a47a57a73be8d060efd9645b4d642f03e
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-09-08 09:46:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 3064469c073d -r 13282d7a47a5 gfx/angle/checkout/src/gpu_info_util/SystemInfo.h
--- a/gfx/angle/checkout/src/gpu_info_util/SystemInfo.h	Sat Sep 07 13:36:46 2019 +0000
+++ b/gfx/angle/checkout/src/gpu_info_util/SystemInfo.h	Sat Sep 07 13:38:36 2019 +0000
@@ -52,27 +52,29 @@
 
     SystemInfo(const SystemInfo &other);
 
+    bool hasNVIDIAGPU() const;
+    bool hasIntelGPU() const;
+    bool hasAMDGPU() const;
+
     std::vector<GPUDeviceInfo> gpus;
 
-    // Index of the primary GPU (the discrete one on dual GPU systems) in `gpus`.
-    // Will never be -1 after a successful GetSystemInfo.
-    int primaryGPUIndex = -1;
-    // Index of the currently active GPU in `gpus`, can be -1 if the active GPU could not be
-    // detected.
+    // Index of the GPU expected to be used for 3D graphics. Based on a best-guess heuristic on
+    // some platforms. On windows, this is accurate.
     int activeGPUIndex = -1;
 
     bool isOptimus       = false;
     bool isAMDSwitchable = false;
+    // Only true on dual-GPU Mac laptops.
+    bool isMacSwitchable = false;
 
-    // Only available on Android, when added by the feature support utility
+    // Only available on Android
     std::string machineManufacturer;
 
+    // Only available on macOS and Android
+    std::string machineModelName;
+
     // Only available on macOS
-    std::string machineModelName;
     std::string machineModelVersion;
-
-    // Only available on Windows, set even on failure.
-    std::string primaryDisplayDeviceId;
 };
 
 // Gathers information about the system without starting a GPU driver and returns them in `info`.
@@ -85,8 +87,9 @@
 constexpr VendorID kVendorID_ARM      = 0x13B5;
 constexpr VendorID kVendorID_ImgTec   = 0x1010;
 constexpr VendorID kVendorID_Intel    = 0x8086;
-constexpr VendorID kVendorID_Nvidia   = 0x10DE;
+constexpr VendorID kVendorID_NVIDIA   = 0x10DE;
 constexpr VendorID kVendorID_Qualcomm = 0x5143;
+constexpr VendorID kVendorID_VMWare   = 0x15ad;
 
 // Known non-PCI (i.e. Khronos-registered) vendor IDs
 constexpr VendorID kVendorID_Vivante     = 0x10001;
@@ -99,11 +102,14 @@
 bool IsImgTec(VendorID vendorId);
 bool IsIntel(VendorID vendorId);
 bool IsKazan(VendorID vendorId);
-bool IsNvidia(VendorID vendorId);
+bool IsNVIDIA(VendorID vendorId);
 bool IsQualcomm(VendorID vendorId);
 bool IsVeriSilicon(VendorID vendorId);
+bool IsVMWare(VendorID vendorId);
 bool IsVivante(VendorID vendorId);
 
+// Dumps the system info to stdout.
+void PrintSystemInfo(const SystemInfo &info);
 }  // namespace angle
 
 #endif  // GPU_INFO_UTIL_SYSTEM_INFO_H_